machine/v1beta1/types_awsprovider.go

```diff
@@ -393,3 +401,46 @@ const (
 	// When set to CapacityBlock the instance utilizes pre-purchased compute capacity (capacity blocks) with AWS Capacity Reservations.
 	MarketTypeCapacityBlock MarketType = "CapacityBlock"
 )
+
+// HostPlacement is the type that will be used to configure the placement of AWS instances.
+// +kubebuilder:validation:XValidation:rule="has(self.type) && self.affinity == 'DedicatedHost' ?  has(self.dedicated) : !has(self.dedicated)",message="dedicated is required when affinity is DedicatedHost, and forbidden otherwise"
+// +union
+type HostPlacement struct {
+	// affinity specifies the affinity setting for the instance.
+	// Allowed values are AnyAvailable and DedicatedHost
+	// When Affinity is set to DedicatedHost, an instance started onto a specific host always restarts on the same host if stopped. In this scenario, the `dedicated` field must be set.
+	// When Affinity is set to AnyAvailable, and you stop and restart the instance, it can be restarted on any available host.
+	// +required
+	// +unionDiscriminator
+	Affinity HostAffinity `json:"affinity,omitempty"`
```

Comment:
Because this is validated in a go webhook, do you need this to be a pointer so you can explicitly distinguish between not set and intentionally set to the empty string value (`""`) and return the appropriate field error (i.e required vs invalid value)?